ignite/handlers/param_scheduler.py

```diff
@@ -325,18 +339,33 @@ def __init__(
             "cycle",
             "start_value_mult",
             "end_value_mult",
+            "total_cycle_size",
+            "warmup_duration",
         ]
 
     def __call__(self, engine: Optional[Engine], name: Optional[str] = None) -> None:
-        if self.event_index != 0 and self.event_index % self.cycle_size == 0:
+        if self.event_index != 0 and self.event_index % self.total_cycle_size == 0:
             self.event_index = 0
             self.cycle_size = int(self.cycle_size * self.cycle_mult)
+            self.warmup_duration = int(self.warmup_duration * self.cycle_mult)
+            self.total_cycle_size = int(self.warmup_duration + self.cycle_size)
             self.cycle += 1
             self.start_value *= self.start_value_mult
+        if self.event_index != 0 and self.event_index == self.warmup_duration:
             self.end_value *= self.end_value_mult
 
         return super(CyclicalScheduler, self).__call__(engine, name)
 
+    def get_param(self) -> float:
+        """Method to get current optimizer's parameter value"""
+        if self.warmup_each_cycle and self.event_index < self.warmup_duration:
+            return self.end_value + (self.start_value - self.end_value) * self.event_index / self.warmup_duration
+
+        return self._get_cycle_param()
+
+    def _get_cycle_param(self):
```

Comment:
```suggestion
    def _get_cycle_param(self) -> float:
```

Reply:
@sihyeong671 To implement an abstract class please use this pattern: https://github.com/pytorch/ignite/blob/ca3033a7ac30d6afae4eb6f5a03d6a6ffad49c84/ignite/handlers/param_scheduler.py#L89-L96

Reply:
> @sihyeong671 To implement an abstract class please use this pattern:
> 
> https://github.com/pytorch/ignite/blob/ca3033a7ac30d6afae4eb6f5a03d6a6ffad49c84/ignite/handlers/param_scheduler.py#L89-L96

@vfdev-5 @sadra-barikbin 
Thank for review my code!
you mean change below code to above pattern on line89?
```py
def _get_cycle_param(self) -> float:
        """Method to get the cycle's current parameter value"""
        raise NotImplementedError
```